static/app/utils/theme/theme.chonk.tsx

```diff
@@ -830,3 +827,116 @@ declare module '@emotion/react' {
+interface ChonkCreateStyled {
+  <
+    C extends React.ComponentClass<React.ComponentProps<C>>,
+    ForwardedProps extends keyof React.ComponentProps<C> &
+      string = keyof React.ComponentProps<C> & string,
+  >(
+    component: C,
+    options: FilteringStyledOptions<React.ComponentProps<C>, ForwardedProps>
+  ): CreateStyledComponent<
+    Pick<React.ComponentProps<C>, ForwardedProps> & {
+      theme?: DO_NOT_USE_ChonkTheme;
+    },
+    Record<string, unknown>,
+    {
+      ref?: React.Ref<InstanceType<C>>;
+    }
+  >;
+  <C extends React.ComponentClass<React.ComponentProps<C>>>(
+    component: C,
+    options?: StyledOptions<React.ComponentProps<C>>
+  ): CreateStyledComponent<
+    React.ComponentProps<C> & {
+      theme?: DO_NOT_USE_ChonkTheme;
+    },
+    Record<string, unknown>,
+    {
+      ref?: React.Ref<InstanceType<C>>;
+    }
+  >;
+  <
+    C extends React.ComponentType<React.ComponentProps<C>>,
+    ForwardedProps extends keyof React.ComponentProps<C> &
+      string = keyof React.ComponentProps<C> & string,
+  >(
+    component: C,
+    options: FilteringStyledOptions<React.ComponentProps<C>, ForwardedProps>
+  ): CreateStyledComponent<
+    Pick<React.ComponentProps<C>, ForwardedProps> & {
+      theme?: DO_NOT_USE_ChonkTheme;
+    }
+  >;
+  <C extends React.ComponentType<React.ComponentProps<C>>>(
+    component: C,
+    options?: StyledOptions<React.ComponentProps<C>>
+  ): CreateStyledComponent<
+    React.ComponentProps<C> & {
+      theme?: DO_NOT_USE_ChonkTheme;
+    }
+  >;
+  <
+    Tag extends keyof React.JSX.IntrinsicElements,
+    ForwardedProps extends keyof React.JSX.IntrinsicElements[Tag] &
+      string = keyof React.JSX.IntrinsicElements[Tag] & string,
+  >(
+    tag: Tag,
+    options: FilteringStyledOptions<React.JSX.IntrinsicElements[Tag], ForwardedProps>
+  ): CreateStyledComponent<
+    {
+      as?: React.ElementType;
+      theme?: DO_NOT_USE_ChonkTheme;
+    },
+    Pick<React.JSX.IntrinsicElements[Tag], ForwardedProps>
+  >;
+  <Tag extends keyof React.JSX.IntrinsicElements>(
+    tag: Tag,
+    options?: StyledOptions<React.JSX.IntrinsicElements[Tag]>
+  ): CreateStyledComponent<
+    {
+      as?: React.ElementType;
+      theme?: DO_NOT_USE_ChonkTheme;
+    },
+    React.JSX.IntrinsicElements[Tag]
+  >;
+}
+
+type ChonkStyled = {
+  [Tag in keyof React.JSX.IntrinsicElements]: CreateStyledComponent<
+    {
+      as?: React.ElementType;
+      theme?: DO_NOT_USE_ChonkTheme;
+    },
+    React.JSX.IntrinsicElements[Tag]
+  >;
+};
+
+// Emotion has no override available for styled, so we create our own,
+// which allows us to use chonkStyled and access the chonk theme and write
+// our components with a future type API.
+interface ChonkStyle extends ChonkCreateStyled, ChonkStyled {}
+export const chonkStyled = styled as ChonkStyle;
+
+export function useChonkTheme(): ChonkTheme {
+  const theme = useTheme() as Theme | ChonkTheme;
+
+  assertChonkTheme(theme);
+  return theme as ChonkTheme;
```

Comment:
this type assertion is unnecessary because `assertChonkTheme` does type narrowing:

```suggestion
  return theme;
```